api/prisma/migrateToVersionedPublications.ts

```diff
@@ -1,3 +1,5 @@
+// Disable TS checking as this is only used against an outdated prisma schema
+// @ts-nocheck
```

Comment:
Disabling TS checking on this file can lead to data loss while migrating data from one model to another, especially for optional fields and relations. I think it would have been better/safer to rename the old Publication model first, then create the new models so that we have TS type checking, autocomplete etc for the old model and for the new models as well. Then we can create new publications and their v1 while looping through the old ones and mapping all the necessary fields/relations. After that we can simply drop the old (renamed) Publication model and it's relations.

Reply:
What I could do is change the migration instructions to instruct that only the specific commit is checked out, then TS checking will be enabled at the point the script is run (this was the case when developing and testing the migration script, so we know the TS checking works). After the "versioning_post_migration" prisma migration is run, we wouldn't have any need for this script. It may actually be better to remove the script entirely instead of marking it as ts-nocheck. I'll do that instead and edit the migration instructions